custom_components/inpost_paczkomaty/api.py

```diff
@@ -1,154 +1,376 @@
+        if response.is_error:
+            _LOGGER.error("API request failed with status %d", response.status)
             raise ApiClientError(
-                f"Error communicating with API! Status: {e.status}"
-            ) from e
+                f"Error communicating with InPost API! Status: {response.status}"
+            )
+
+        # Convert camelCase keys to snake_case
+        converted_data = convert_keys_to_snake_case(response.body)
+
+        # Parse response using dacite
+        dacite_config = Config(
+            type_hooks={
+                ApiLocation: lambda d: from_dict(ApiLocation, d, config=Config()),
+                ApiAddressDetails: lambda d: from_dict(
+                    ApiAddressDetails, d, config=Config()
+                ),
+                ApiPickUpPoint: lambda d: from_dict(ApiPickUpPoint, d, config=Config()),
+                ApiPhoneNumber: lambda d: from_dict(ApiPhoneNumber, d, config=Config()),
+                ApiReceiver: lambda d: from_dict(ApiReceiver, d, config=Config()),
+                ApiSender: lambda d: from_dict(ApiSender, d, config=Config()),
+            }
+        )
+        tracked_response = from_dict(
+            TrackedParcelsResponse, converted_data, config=dacite_config
+        )
 
-        except TimeoutError as e:
-            _LOGGER.warning("Request timed out")
-            raise ApiClientError("Request timed out") from e
-        except Exception as exception:  # pylint: disable=broad-except
-            _LOGGER.warning("Unknown API error occurred: %s", exception)
+        return self._build_parcels_summary(tracked_response.parcels)
 
-            raise ApiClientError("Error communicating with API!") from exception
+    async def get_profile(self) -> UserProfile:
+        """Get user profile with favorite lockers.
 
+        Returns:
+            UserProfile with delivery points and personal info.
 
-class InPostApi:
-    def __init__(self, hass: HomeAssistant) -> None:
-        """Init class."""
-        self.hass = hass
-        self.session = async_create_clientsession(hass)
+        Raises:
+            ApiClientError: If API request fails.
+        """
+        await self._ensure_valid_token()
 
-    async def _request(
-        self,
-        method: str,
-        url: str,
-        headers: dict | None = None,
-    ) -> ClientResponse:
-        """Get information from the API."""
-        try:
-            async with asyncio.timeout(30):
-                response = await self.session.request(
-                    method=method,
-                    url=url,
-                    headers=headers,
-                )
-                response.raise_for_status()
+        response = await self._http_client.get(
+            url=f"{API_BASE_URL}{self.PROFILE_ENDPOINT}",
+            custom_headers={
+                # without this InPost API returns 500 Internal Server Error
+                "User-Agent": "InPost-Mobile/4.4.2 (1)-release (iOS 26.2; iPhone15,3; pl)",
```

Comment:
Could you please move it to CONST? This is repeated a second time in this file and we will need to bump it often I guess.